packages/p2p/src/hooks/__tests__/use-p2p-rendered-adverts.spec.tsx

```diff
@@ -0,0 +1,327 @@
+import React from 'react';
+import { renderHook } from '@testing-library/react-hooks';
+import { mockStore, StoreProvider } from '@deriv/stores';
+import { TCoreStores } from '@deriv/stores/types';
+import { useStores } from 'Stores/index';
+import useP2PRenderedAdverts from '../use-p2p-rendered-adverts';
+
+jest.mock('@deriv/hooks', () => ({
+    ...jest.requireActual('@deriv/hooks'),
+    useP2PAdvertList: jest.fn(() => ({
+        data: [
+            {
+                account_currency: 'USD',
+                advertiser_details: {
+                    name: 'client Test90000253',
+                },
+                counterparty_type: 'buy',
+                type: 'buy',
+                country: 'id',
+                description: 'Created by script. Please call me 02203400',
+            },
+            {
+                account_currency: 'USD',
+                advertiser_details: {
+                    name: 'client Test90000254',
+                },
+                counterparty_type: 'sell',
+                type: 'sell',
+                country: 'id',
+                description: 'Created by script. Please call me 02203400',
+            },
+        ],
+    })),
+}));
+jest.mock('Stores', () => ({
+    ...jest.requireActual('Stores'),
+    useStores: jest.fn().mockReturnValue({}),
+}));
+
+const mockUseStores = useStores as jest.MockedFunction<typeof useStores>;
+
+const renderHookWithConfig = (config: Record<string, object>, mock?: TCoreStores) => {
+    const mock_store = mock || mockStore({ ui: { is_mobile: false } });
+    mockUseStores.mockReturnValue(config);
+    const wrapper = ({ children }: { children: JSX.Element }) => (
+        <StoreProvider store={mock_store}>{children}</StoreProvider>
+    );
+    const { result } = renderHook(() => useP2PRenderedAdverts(), { wrapper });
+    return result.current.rendered_adverts;
+};
+
+describe('useP2PRenderedAdverts', () => {
+    it('should return the adverts object from response with type sell when table_type is buy', () => {
+        const config = {
+            general_store: {
+                lisst_item_limit: 50,
```

Comment:
Can we rename this 👀 

```suggestion
                list_item_limit: 50,
```